Sets up the basic structure of the application

pboothe opened the pull request:
Parses command-line arguments
Sets up the basic structure of the application
Adds support for finding cyclic signals in our data

Looks like a very large pull request mostly because the LICENSE file is so big.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pboothe/signal-searcher/2)
<!-- Reviewable:end -->

fernandalavalle commented:
Review status: 0 of 9 files reviewed at latest revision, 5 unresolved discussions.

---

*[cyclic.py, line 44 \[r1\]](https://reviewable.io:443/reviews/pboothe/signal-searcher/2#-KOfoeWasw4n81XObGDg:-KOfoeWbUhYY0mHcPCle:b5tjduz) ([raw file](https://github.com/pboothe/signal-searcher/blob/52ba83e9ba1fe2c08d26fd736cccc0f0b5832599/cyclic.py#L44)):*
> ```Python
>   Args:
>       timeseries: The timeseries in question, one datapoint per hour, in order
> ```

can we document more clearly how time series are going to be represented? I assume an array?

---

*[cyclic.py, line 46 \[r1\]](https://reviewable.io:443/reviews/pboothe/signal-searcher/2#-KOfpfird281yhfV4lOV:-KOfpfis4vi5aJE7XAOd:bs5xzem) ([raw file](https://github.com/pboothe/signal-searcher/blob/52ba83e9ba1fe2c08d26fd736cccc0f0b5832599/cyclic.py#L46)):*
> ```Python
> 
>   Returns:
>       Whether or not a strong 24-hour signal was found
> ```

This is a little nit-picky, but can we explicity say that a boolean will be returned?

---

*[cyclic.py, line 76 \[r1\]](https://reviewable.io:443/reviews/pboothe/signal-searcher/2#-KOjZHCegihGxY0pk9lZ:-KOjZHCegihGxY0pk9l_:b-f9zbx2) ([raw file](https://github.com/pboothe/signal-searcher/blob/52ba83e9ba1fe2c08d26fd736cccc0f0b5832599/cyclic.py#L76)):*
> ```Python
>   problems_found = []
>   for netblock in timeseries:
>     problem_size = netblock_size(netblock) * (len(timeseries) / 24.0)
> ```

I'm not sure what the problem size refers to? Can you add a clarifying note?

---

*[cyclic_test.py, line 65 \[r1\]](https://reviewable.io:443/reviews/pboothe/signal-searcher/2#-KOj_Jd4JEe9kO0Nas6w:-KOj_Jd4JEe9kO0Nas6x:b-lvk0o3) ([raw file](https://github.com/pboothe/signal-searcher/blob/52ba83e9ba1fe2c08d26fd736cccc0f0b5832599/cyclic_test.py#L65)):*
> ```Python
>   def test_short_cyclic_data_has_many_problems(self):
>     self.assertNotEqual(len(cyclic.find_problems(self.short_cyclic_data)), 0)
> ```

Should we write some tests that check if the specific problem that is present is caught?

---

*[mlabreader.py, line 33 \[r1\]](https://reviewable.io:443/reviews/pboothe/signal-searcher/2#-KOjcCkzAqCyYWhl3ZWe:-KOjcCkzAqCyYWhl3ZWf:bmjo5p7) ([raw file](https://github.com/pboothe/signal-searcher/blob/52ba83e9ba1fe2c08d26fd736cccc0f0b5832599/mlabreader.py#L33)):*
> ```Python
>       data[block].append(
>           MlabDataEntry(
>               time=t, upload_speed=5, download_speed=10, min_latency=100))
> ```

I feel like I'm definitely missing something, why are we setting the values of the metrics? shouldn't we be reading in data?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pboothe/signal-searcher/2)*
<!-- Sent from Reviewable.io -->

pboothe commented:
Review status: 0 of 9 files reviewed at latest revision, 5 unresolved discussions.

---

*[cyclic.py, line 44 \[r1\]](https://reviewable.io:443/reviews/pboothe/signal-searcher/2#-KOfoeWasw4n81XObGDg:-KOjoHaVZ6Rb695xd4gz:b-s1iwpd) ([raw file](https://github.com/pboothe/signal-searcher/blob/52ba83e9ba1fe2c08d26fd736cccc0f0b5832599/cyclic.py#L44)):*
<details><summary><i>Previously, fernandalavalle wrote…</i></summary>
> can we document more clearly how time series are going to be represented? I assume an array?
</details>
I have documented that it should be a list, which is the python way of saying array.

---

*[cyclic.py, line 46 \[r1\]](https://reviewable.io:443/reviews/pboothe/signal-searcher/2#-KOfpfird281yhfV4lOV:-KOjp3pOoaeZpq1VzjHb:b-896fix) ([raw file](https://github.com/pboothe/signal-searcher/blob/52ba83e9ba1fe2c08d26fd736cccc0f0b5832599/cyclic.py#L46)):*
<details><summary><i>Previously, fernandalavalle wrote…</i></summary>
> This is a little nit-picky, but can we explicity say that a boolean will be returned?
</details>
Done.

---

*[cyclic.py, line 76 \[r1\]](https://reviewable.io:443/reviews/pboothe/signal-searcher/2#-KOjZHCegihGxY0pk9lZ:-KOjrI_lss94xaBfMXLu:b-896fix) ([raw file](https://github.com/pboothe/signal-searcher/blob/52ba83e9ba1fe2c08d26fd736cccc0f0b5832599/cyclic.py#L76)):*
<details><summary><i>Previously, fernandalavalle wrote…</i></summary>
> I'm not sure what the problem size refers to? Can you add a clarifying note?
</details>
Done.

---

*[cyclic_test.py, line 65 \[r1\]](https://reviewable.io:443/reviews/pboothe/signal-searcher/2#-KOj_Jd4JEe9kO0Nas6w:-KOjrO1MsmjgJ1x8SFK7:b-ko15lb) ([raw file](https://github.com/pboothe/signal-searcher/blob/52ba83e9ba1fe2c08d26fd736cccc0f0b5832599/cyclic_test.py#L65)):*
<details><summary><i>Previously, fernandalavalle wrote…</i></summary>
> Should we write some tests that check if the specific problem that is present is caught?
</details>
Maybe? The issue is that I don't want the tests here to have to be updated whenever the constructor for the reports changes - that seems like needless entanglement between the modules. But maybe they should be?

I have updated the tests to be a little more rigorous and to make that that each list of problems, rather than simply being non-zero, actually has the right number of problems discovered.  Also, I refactored the tests object because `setUp()` did more than it had to and needlessly created variables with a larger scope.  Hopefully this way we can meet in the middle.

---

*[mlabreader.py, line 33 \[r1\]](https://reviewable.io:443/reviews/pboothe/signal-searcher/2#-KOjcCkzAqCyYWhl3ZWe:-KOjrla76YapJzZYC4Dm:b7wv9qt) ([raw file](https://github.com/pboothe/signal-searcher/blob/52ba83e9ba1fe2c08d26fd736cccc0f0b5832599/mlabreader.py#L33)):*
<details><summary><i>Previously, fernandalavalle wrote…</i></summary>
> I feel like I'm definitely missing something, why are we setting the values of the metrics? shouldn't we be reading in data?
</details>
Yes! This is a dummy implementation, to be replaced by the thing you are writing. I have added documentation it to that effect to make it clear that this function is a placeholder for your work.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pboothe/signal-searcher/2)*
<!-- Sent from Reviewable.io -->

fernandalavalle commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 9 files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pboothe/signal-searcher/2#-:-KOjyIGNXSLb0QtKYb8O:bnfp4nl)*
<!-- Sent from Reviewable.io -->
